synapse/rest/client/account.py

```diff
@@ -299,19 +299,16 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
 
         requester = await self.auth.get_user_by_req(request)
 
-        # allow ASes to deactivate their own users
-        if requester.app_service:
-            await self._deactivate_account_handler.deactivate_account(
-                requester.user.to_string(), body.erase, requester
+        # allow ASes to deactivate their own users:
```

Comment:
This somehow made me think we should be checking if the appservice owns the user, but `requester.app_service` wouldn't exist unless the AS owned the user.